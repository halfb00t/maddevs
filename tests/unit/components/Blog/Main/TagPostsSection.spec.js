import { render } from '@testing-library/vue'
import { mount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import TagPostsSection from '@/components/Blog/Main/TagPostsSection'
import tagPosts from '../../../__mocks__/tagPosts'
import initializeLazyLoad from '@/helpers/lazyLoad'

const localVue = createLocalVue()

localVue.use(Vuex)

const mocks = {
  $prismic: {
    asText: () => 'img.png',
  },
}

const store = {
  getters: {
    blogTag: () => 'Blog tag',
    tagPostsCount: () => 5,
    tagPostsLoaded: () => true,
    allAuthors: () => [
      {
        id: '1',
      }, {
        id: '2',
      }, {
        id: '3',
      },
    ],
    tagPostsPage: () => 1,
    tagPosts: () => tagPosts,
  },
}

const stubs = ['FeaturedPost', 'NuxtLink', 'PostCard']

jest.mock('@/helpers/lazyLoad', () => ({
  initializeLazyLoad: jest.fn(),
}))

describe('TagPostsSection component', () => {
  it('should render correctly', () => {
    const { container } = render(TagPostsSection, {
      store,
      mocks,
      stubs,
    })

    expect(container).toMatchSnapshot()
  })

  it('if tagPosts is empty > will return null from featuredPostAuthor', () => {
    const wrapper = mount(TagPostsSection, {
      localVue,
      mocks,
      stubs,
      store: {
        getters: {
          blogTag: () => 'Blog tag',
          tagPostsCount: () => 5,
          tagPostsLoaded: () => true,
          allAuthors: () => [
            {
              id: '1',
            }, {
              id: '2',
            }, {
              id: '3',
            },
          ],
          tagPostsPage: () => 1,
          tagPosts: [],
        },
      },
    })
    expect(wrapper.vm.featuredPostAuthor).toBeNull()
  })

  it('if tagPosts has data > will return { id: "YG83_xAAACIA9tnb", name: "Denisoed" }', async () => {
    const initLazyLoadMock = jest.spyOn(initializeLazyLoad, 'initializeLazyLoad').mockImplementation(() => {})
    const nextTick = jest.fn(() => initLazyLoadMock())
    mocks.$nextTick = nextTick
    const wrapper = mount(TagPostsSection, {
      localVue,
      mocks,
      stubs,
      store: {
        getters: {
          blogTag: 'Blog tag',
          tagPostsCount: 5,
          tagPostsLoaded: true,
          allAuthors: [
            {
              id: 'YG83_xAAACIA9tnb',
              name: 'Denisoed',
            }, {
              id: '2',
              name: 'Baha',
            }, {
              id: '3',
              name: 'Ivan',
            },
          ],
          tagPostsPage: 1,
          tagPosts,
        },
      },
    })
    expect(wrapper.vm.featuredPostAuthor).toEqual({ id: 'YG83_xAAACIA9tnb', name: 'Denisoed' })
    await wrapper.vm.$forceUpdate()
    await wrapper.vm.$nextTick(() => {
      expect(initLazyLoadMock).toHaveBeenCalledTimes(1)
    })
  })
})
