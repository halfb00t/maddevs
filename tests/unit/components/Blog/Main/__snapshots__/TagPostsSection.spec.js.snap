// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagPostsSection component should render correctly 1`] = `
<div>
  <div
    class="tag-posts"
  >
    <div
      class="container"
    >
      <div
        class="tag-posts__featured-post"
      >
        <div
          class="featured-post light"
        >
          <div
            class="featured-post__wrapper"
          >
            <div
              class="featured-post__main"
            >
              <h2
                class="featured-post__title background-grey animated light"
              />
               
              <p
                class="featured-post__paragraph background-grey animated light"
              />
               
              <div
                class="featured-post__data d-flex justify-content-between"
              >
                <div
                  class="featured-post__meta"
                >
                  <span
                    class="created-at"
                  />
                </div>
              </div>
            </div>
             
            <div
              class="featured-post__image-wrapper"
            >
              <div
                class="featured-post__image background-grey animated light"
              />
            </div>
          </div>
        </div>
      </div>
       
      <div
        class="tag-posts__list"
      >
        <section
          class="tag-posts__list-item"
          post="[object Object]"
        >
          <postcard-stub
            disabletaglink="true"
            post="[object Object]"
            tag="Blog tag"
            theme="light"
          />
        </section>
        <section
          class="tag-posts__list-item"
          post="[object Object]"
        >
          <postcard-stub
            disabletaglink="true"
            post="[object Object]"
            tag="Blog tag"
            theme="light"
          />
        </section>
        <section
          class="tag-posts__list-item"
          post="[object Object]"
        >
          <postcard-stub
            disabletaglink="true"
            post="[object Object]"
            tag="Blog tag"
            theme="light"
          />
        </section>
      </div>
       
      <!---->
    </div>
  </div>
</div>
`;
